src/python/pants/backend/python/util_rules/local_dists_pep660.py

```diff
@@ -0,0 +1,430 @@
+      https://peps.python.org/pep-0660/
+      https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/
+      https://packaging.python.org/en/latest/specifications/recording-installed-packages/
+      https://packaging.python.org/en/latest/specifications/direct-url-data-structure/
+      https://packaging.python.org/en/latest/specifications/binary-distribution-format/
+    """
+
+    # Create the .pth files to add the relevant source root to sys.path.
+    # We cannot use the build backend to do this because we do not want to tell
+    # it where the workspace is and risk it adding anything there.
+    # NOTE: We use .pth files to support ICs less than python3.7.
+    #       A future enhancement might be to provide more precise editable
+    #       wheels based on https://pypi.org/project/editables/, but that only
+    #       supports python3.7+ (what pip supports as of April 2023).
+    #       Or maybe do something like setuptools strict editable wheel.
+    pth_file_contents = ""
+    direct_url = ""
+    for source_root in request.build_time_source_roots:
+        # can we use just the first one to only have the dist's source root?
+        abs_path = (
+            build_root.path if source_root == "." else str(build_root.pathlib_path / source_root)
+        )
+        pth_file_contents += f"{abs_path}\n"
+        if not direct_url:  # use just the first source_root
+            direct_url = "file://" + abs_path.replace(os.path.sep, "/")
+    pth_file_name = "__pants__.pth"
+    pth_file_path = os.path.join(request.working_directory, pth_file_name)
+
+    # This is the setuptools dist directory, not Pants's, so we hardcode to dist/.
+    dist_dir = "dist"
+    dist_output_dir = os.path.join(dist_dir, request.output_path)
+
+    backend_wrapper_json = "backend_wrapper.json"
+    backend_wrapper_json_path = os.path.join(request.working_directory, backend_wrapper_json)
+    backend_wrapper_name = "backend_wrapper.py"
+    backend_wrapper_path = os.path.join(request.working_directory, backend_wrapper_name)
+    backend_wrapper_content = read_resource(_scripts_package, "pep660_backend_wrapper.py")
+    assert backend_wrapper_content is not None
+
+    conf_digest, backend_wrapper_digest, build_backend_pex = await MultiGet(
+        Get(
+            Digest,
+            CreateDigest(
+                [
+                    FileContent(pth_file_path, pth_file_contents.encode()),
+                    FileContent(
+                        backend_wrapper_json_path,
+                        dump_backend_wrapper_json(
+                            dist_output_dir, pth_file_name, direct_url, request
+                        ),
+                    ),
+                ]
+            ),
+        ),
+        # The backend_wrapper has its own digest for cache reuse.
+        Get(
+            Digest,
+            CreateDigest([FileContent(backend_wrapper_path, backend_wrapper_content)]),
+        ),
```

Comment:
Any reason this isn't part of the previous `CreateDigest`?

Reply:
and now I see the comment--lol :p

Reply:
I wonder how big impact it has, if any, though.. curious what @stuhood thinks.

Reply:
I don't know how much of an impact it has. For ST2 we have 18 dists, each of which will have a different set of settings to pass in (in the `conf_digest`) but the `backend_wrapper` is the same for all of them. So, I figured I'd split the 2 digests up here.

We do merge the digests below, but we would have to do that merge anyway to include any input digest.

I can combine them. I don't know what impact it has one way or another, but I suspect this is at least marginally better.

Reply:
Yea I've no clue either. Guess it'll comer down to the performance of materializing an extra file in a digest vs running an additional (with cached result) workunit.